saleor/graphql/shop/types.py

```diff
@@ -352,8 +353,17 @@ class Meta:
         description = (
             "Represents a shop resource containing general shop data and configuration."
         )
-        interfaces = [ObjectWithMetadata]
-
+        interfaces = [graphene.relay.Node,ObjectWithMetadata]
+        model = site_models.SiteSettings
+    
+    @staticmethod
+    def resolve_id(_, _info):
+        return "1"
```

Comment:
It should return Base64 encoded string "Shop:1".

Reply:
I think this should work:
```
@staticmethod
def resolve_id(_, _info):
    return base64.b64encode(b"Shop:1").decode()
```

Reply:
It's not working . Unable to delete.

Reply:
It's working with return "1". Returning string like `U2hvcDox`

Reply:
The resolver needs to return Base64 encoded version. To support deleting and other metadata mutations I think we also need to make changes to `BaseMetadataMutation`; not sure now which methods need changing but there is the entire logic for resolving objects for these mutations. Since we have a custom ID for the shop object, I think we need to adjust some of the methods in `BaseMetadataMutation` to properly translate from `U2hvcDox` to `SiteSettings.objects.first()`. For example this mutation should work and remove metadata from the Shop:

```
mutation {
  deleteMetadata(id: "U2hvcDox", keys: ["test"]) {
    errors {
      message
      field
    }
  }
}
```

Reply:
Above mutation is working .